### ISSUE TITLE
Add websocket support

### ISSUE THREAD

**Issue by flyswatter** [MetaMask/metamask-plugin#1645]
Create a branch to work with Infura Websocket support.

Our websocket provider would not need to handle filters client side.

Won't need client-side filter management in the short term, for experimenting with.

**Comment by danfinlay** [comment]
Raising priority because it will allow a tighter integration with ganache to detect provider changes. #1999.

**Comment by bdresser** [comment]
relates to #4279

**Comment by lazaridiscom** [comment]
please notice https://github.com/MetaMask/web3-provider-engine/issues/189#issuecomment-557033566

**Comment by d10r** [comment]
So, websocket connections aren't yet supported in Metamask, correct?
I'm currently using a websocket RPC connection (readonly) in some Dapps, in parallel to the Metamask provided connection, in order to have the UI react faster to on-chain changes.

From this issue, it's not clear to me if websocket support is to be expected anytime soon. That would help me decide if I should treat the extra code for that workaround as temporary, or if I should be prepared to live with it for a bit longer and thus give it some more love.

**Comment by danfinlay** [comment]
It is not yet supported, and it isn't one of our top priorities at the moment, so I would expect to live with other approaches for a bit longer, sorry!

**Comment by lazaridiscom** [comment]
> and it isn't one of our top priorities at the moment

@danfinlay, the P1 should be possibly changed to P2 here, to reflect this.

@d10r and others interested in this:

The difficulty is not that much to implement websocket support, but to implement it in "multiple running targets", e.g. multiple components/subsystems which are both, live and(!) under ongoing development. Used by 100k if not millions of users.

Due to experiences with other mm "mega issues", I'd say that this here can be done in:

- 1 to 3 months planning/research, to determine solution-paths/sequences (one ~80% assigned analyst, very difficult to share/delegate the work)
- 1 to 3 months implementation, testing and rollout (one ~80% assigned core dev (possibly even a pair), can delegate tasks to assisting devs to fasten things up)

I hear now folks screaming "what? this is just this-and-that". Yes, but once you start analyzing the problem domain, you'll most possibly get dizzy within a few days (from all the "moving targets"). 

So, although websockets is possibly the "modern way of doing things", I understand why the devs set this on low priority.

Things work as-is, and daily business/production needs to go on.

**Comment by markg85** [comment]
Any update on this, as we're well over 2 years later since the last comment.

**Comment by testermumatstudio** [comment]
How can I make use of metamask using an email, I'll explain the project a little to see if you can guide me on how I should do it, if I create a video game in unity for playstation how do I connect the metamask it's not possible!! I should authenticate with an email and that it connects to metamask I can think of an application for the intermission, you authenticate with email on the playstation it sends a connection to a mobile app to pass the info to the metamask metamask passes the data aal mail and return to playstation is it possible? how can I do it? can you guide me please I am willing to work on it and upload it opensource

**Comment by EvelinaBuiciag** [comment]
Hi, 
Any updates on this? on adding websocket support in metamask ??

Thank you
